### PR TITLE
Enabled configurable plugin caching for CMS 3.3.0+ installations

### PR DESCRIPTION
**Pull request by mkoistinen**
This PR adds an extra "Cache Duration" field, and uses it to set caching for "time-sensitive" addons if and only if the project is running on CMS 3.3.0+.

### CODE REVIEW

File: aldryn_newsblog/cms_plugins.py [hunk 1]
```diff
@@ -27,11 +29,24 @@ class NewsBlogPlugin(TemplatePrefixMixin, CMSPluginBase):
...
 
+class CacheMixin(object):
+    """
+    For django CMS < 3.3.0 installations, we have no choice but to disable the
+    cache where there is time-sensitive information. However, in later CMS
+    versions, we have other choices, namely the cmsplugin-header-control
+    plugin or similar. This mixin re-enables caching in django CMS >= 3.3.0
+    so that those solutions can be employed.
+    """
+    def __init__(self, *args, **kwargs):
```
Comment (czpython) [inline aldryn_newsblog/cms_plugins.py]: move to class scope

File: aldryn_newsblog/cms_plugins.py [hunk 2]
```diff
@@ -27,11 +31,41 @@ class NewsBlogPlugin(TemplatePrefixMixin, CMSPluginBase):
...
+        using django CMS v3.3.0 or later.
+        """
+        fieldsets = super(AdjustableCacheMixin, self).get_fieldsets(request, obj=None)
+        if CMS_GTE_330:
+            return fieldsets
+
+        field = 'cache_duration'
+        for fieldset in fieldsets:
+            if field in fieldset[1]['fields']:
+                new_fieldset = [
```
Comment (GaretJax) [inline aldryn_newsblog/cms_plugins.py]: You could as well remove the `if`, as the `in` operator causes the fields tuple to be iterated over anyway (so better once than twice).

File: aldryn_newsblog/migrations/0013_auto_20160623_1703.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        migrations.AddField(
+            model_name='newsblogrelatedplugin',
+            name='cache_duration',
+            field=models.PositiveSmallIntegerField(default=0, help_text="The maximum duration (in seconds) that this plugin's contentshould be cached. Values between 5 minutes (300) and one day (84600) recommended)."),
+        ),
+        migrations.AlterField(
+            model_name='newsblogconfig',
+            name='app_data',
+            field=app_data.fields.AppDataField(default='{}', editable=False),
+        ),
```
Comment (GaretJax) [inline aldryn_newsblog/migrations/0013_auto_20160623_1703.py]: Where does this come from?

File: tox.ini [hunk 4]
```diff
@@ -37,14 +42,14 @@ deps =
 
     cms31: django-cms>=3.1,<3.2
     # NOTE: tests will fail if we restrict to 2.7.0 on 3.1
-    cms32: djangocms-text-ckeditor>2.7,<=2.9.3
+    cms31: djangocms-text-ckeditor>2.7,<=2.9.3
```
Comment (GaretJax) [inline tox.ini]: I'll trust you on those. :)

**Comment by GaretJax** [comment]
Left some minor comments, but LGTM

File: aldryn_newsblog/migrations/0013_auto_20160623_1703.py (hunk 3 above)
Comment (mkoistinen) [inline aldryn_newsblog/migrations/0013_auto_20160623_1703.py]: They were missing migrations.

**Pull request merged**